### PR TITLE
[GDPR] Change Email (Local)

### PR DESCRIPTION
**Pull request by wyattjoh**
## Features

New Plugin: **talk-plugin-local-auth**:

- *Email Change*: Allows users to change their existing email address on their account.

## How do I test this PR?

- TBD

### CODE REVIEW

File: plugins/talk-plugin-auth/client/index.js [hunk 1]
```diff
@@ -13,6 +14,7 @@ export default {
   slots: {
     stream: [UserBox, SignInButton, SetUsernameDialog],
     login: [Login],
+    profileHeader: [Profile],
     profileHeader: [ChangeUsername],
```
Comment (alinhle) [inline plugins/talk-plugin-auth/client/index.js]: This seems to overwrite the previous entry?

**Comment by alinhle** [comment]
Also I encountered this error: `GraphQLError: Variable "$username" of required type "String!" was not provided`

File: plugins/talk-plugin-auth/client/index.js (hunk 1 above)
Comment (okbel) [reply plugins/talk-plugin-auth/client/index.js]: It seems to be a merge conflict. Checking!

Comment (okbel) [reply plugins/talk-plugin-auth/client/index.js]: Yes, this is correct. 

```
    stream: [UserBox, SignInButton, SetUsernameDialog],
    login: [Login],
    profileHeader: [Profile],
    profileSettings: [ChangePassword],
```

This PR handles the change of the email too within Profile Component (which handles both username and email changes)

File: .gitignore [hunk 2]
```diff
@@ -51,8 +52,10 @@ plugins/*
 !plugins/talk-plugin-offtopic
 !plugins/talk-plugin-permalink
 !plugins/talk-plugin-profile-data
+!plugins/talk-plugin-profile-settings
```
Comment (alinhle) [inline .gitignore]: We dont have this plugin anymore right?

Comment (kgardnr) [reply .gitignore]: Oh nope! I must've missed that in removing references to it

Comment (kgardnr) [reply .gitignore]: Removed!

**Review by kgardnr — APPROVED** [review]

**Pull request merged**